yt/config.py

```diff
@@ -66,74 +76,132 @@
-                    print(f"Setting {option} to {cp.get(section, option)}")
-        open(_OLD_CONFIG_FILE + ".old", "w").write(f)
-        new_cp.write(open(_OLD_CONFIG_FILE, "w"))
-
-    msg = (
-        "The configuration file {} is deprecated. "
-        "Please migrate your config to {} by running: "
-        "'yt config migrate'"
-    )
-    warnings.warn(msg.format(_OLD_CONFIG_FILE, CURRENT_CONFIG_FILE))
-
-if not os.path.exists(CURRENT_CONFIG_FILE):
-    cp = configparser.ConfigParser()
-    cp.add_section("yt")
+OLD_CONFIG_FILE = os.path.join(CONFIG_DIR, "ytrc")
+GLOBAL_CONFIG_FILE = os.path.join(CONFIG_DIR, "yt.toml")
+LOCAL_CONFIG_FILE = None
+
+if os.path.exists(OLD_CONFIG_FILE):
+    if os.path.exists(GLOBAL_CONFIG_FILE):
+        msg = (
+            "The configuration file {} is deprecated. "
+            "Please manually remove it to suppress this warning."
+        )
+        warnings.warn(msg.format(OLD_CONFIG_FILE, GLOBAL_CONFIG_FILE))
+    else:
+        msg = (
+            "The configuration file {} is deprecated. "
+            "Please migrate your config to {} by running: "
+            "'yt config migrate'"
+        )
+        warnings.warn(msg.format(OLD_CONFIG_FILE, GLOBAL_CONFIG_FILE))
+        sys.exit(1)
+
+
+if not os.path.exists(GLOBAL_CONFIG_FILE):
+    cfg = {"yt": {}}
     try:
-        with open(CURRENT_CONFIG_FILE, "w") as new_cfg:
-            cp.write(new_cfg)
+        with open(GLOBAL_CONFIG_FILE, mode="w") as fd:
+            toml.dump(cfg, fd)
     except IOError:
         warnings.warn("unable to write new config file")
 
 
-class YTConfigParser(configparser.ConfigParser):
-    def __setitem__(self, key, val):
-        self.set(key[0], key[1], val)
+class YTConfig:
+    def __init__(self, defaults=None):
+        if defaults is None:
+            defaults = {}
+        self.config_root = ConfigNode(None)
+
+    def get(self, section, *keys, callback=None, **kwargs):
+        if callback is None:
+            callback = lambda leaf: leaf.value  # noqa: E731
+
+        return self.config_root.get_leaf(section, *keys, callback=callback)
+
+    def get_most_specific(self, section, *keys, **kwargs):
+        use_fallback = "fallback" in kwargs
+        fallback = kwargs.pop("fallback", None)
+        try:
+            return self.config_root.get_deepest_leaf(section, *keys)
+        except KeyError as e:
+            if use_fallback:
+                return fallback
+            else:
+                raise e
+
+    def update(self, new_values, metadata=None):
+        if metadata is None:
+            metadata = {}
+        self.config_root.update(new_values, metadata)
+
+    def has_section(self, section):
+        try:
+            self.config_root.get_child(section)
+            return True
+        except KeyError:
+            return False
+
+    def add_section(self, section):
+        self.config_root.add_child(section)
+
+    def remove_section(self, section):
+        if self.has_section(section):
```

Comment:
I'm not sure why this check is necessary ? It seems that it could be making some errors silent for no good reason I can think of.

Reply:
Once again, this is merely for consistency with `configparser.ConfigParser.remove_section`. I noticed it returns True or False depending on whether the section was removed successfully, so updating accordingly.